Source/ACE.DatLoader/DatReader.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.IO;
+
+namespace ACE.DatLoader
+{
+    public class DatReader
+    {
+        public int Offset { get; set; }
+        private byte[] buffer;
+
+        public DatReader(string datFilePath, uint offset, uint size, uint sectorSize)
+        {
+            uint nextAddress = 0;
+            FileStream stream = new FileStream(datFilePath, FileMode.Open, FileAccess.Read);
+            using (stream)
+            {
+                this.buffer = new byte[size];
+                stream.Seek(offset, SeekOrigin.Begin);
+
+                if (size > sectorSize)

```

Comment:
So, I know what this does because I wrote the original DAT stuff, but this is totally non-obvious to someone who didn't.  Just put a comment as to why this is the right thing to do.